src/ast.rs

```diff
@@ -1409,6 +1409,28 @@ pub fn create_call_to(
     }
 }
 
+pub fn create_call_to_with_ids(
+    function_name: String,
+    parameters: Vec<AstStatement>,
+    parameter_list_id: usize,
+    location: &SourceRange,
+    mut id_provider: IdProvider,
+) -> AstStatement {
+    AstStatement::CallStatement {
+        operator: Box::new(AstStatement::Reference {
+            name: function_name,
+            location: location.clone(),
+            id: id_provider.next_id(),
+        }),
+        parameters: Box::new(Some(AstStatement::ExpressionList {
+            expressions: parameters,
+            id: parameter_list_id,
```

Comment:
here we can also use a new id and get rid of this parameter_list_id parameter.
or is it important that we reuse the id here?